spokestack/nlu/parsers/maps.py

```diff
@@ -0,0 +1,130 @@
+"""
+This module contains mappings to convert numbers in words
+to integers. ie. five -> 5
+"""
+
+from typing import Dict
+
+
+ENG_ZERO: Dict[str, int] = {
+    "zero": 0,
+    "oh": 0,
+    "owe": 0,
+}
+ENG_MOD10: Dict[str, int] = {
+    "one": 1,
+    "won": 1,
+    "two": 2,
+    "too": 2,
+    "to": 2,
+    "three": 3,
+    "four": 4,
+    "for": 4,
+    "fore": 4,
+    "five": 5,
+    "six": 6,
+    "sicks": 6,
+    "sics": 6,
+    "seven": 7,
+    "eight": 8,
+    "ate": 8,
+    "nine": 9,
+}
+ENG_MOD20: Dict[str, int] = {
+    "ten": 10,
+    "tin": 10,
+    "eleven": 11,
+    "twelve": 12,
+    "thirteen": 13,
+    "fourteen": 14,
+    "fifteen": 15,
+    "sixteen": 16,
+    "seventeen": 17,
+    "eighteen": 18,
+    "nineteen": 19,
+}
+
+ENG_DIV10: Dict[str, int] = {
+    "twenty": 2,
+    "thirty": 3,
+    "forty": 4,
+    "fifty": 5,
+    "sixty": 6,
+    "seventy": 7,
+    "eighty": 8,
+    "ninety": 9,
+}
+
+ENG_EXP10: Dict[str, int] = {"hundred": 2, "thousand": 3}
+
+MULTIPLIERS: Dict[str, int] = {
+    "hundred": 100,
+    "thousand": 1000,
+    "million": int(1e6),
+    "billion": int(1e9),
+}
+WORD_TO_NUM: Dict[str, int] = {
+    "oh": 0,
+    "owe": 0,
+    "zero": 0,
+    "won": 1,
+    "one": 1,
+    "first": 1,
+    "to": 2,
+    "too": 2,
+    "two": 2,
+    "second": 2,
+    "three": 3,
+    "third": 3,
+    "for": 4,
+    "fore": 4,
+    "four": 4,
+    "five": 5,
+    "fif": 5,
+    "sicks": 6,
+    "sics": 6,
+    "six": 6,
+    "seven": 7,
+    "ate": 8,
+    "eight": 8,
+    "eigh": 8,
+    "nine": 9,
+    "nin": 9,
+    "tin": 10,
+    "ten": 10,
+    "eleven": 11,
+    "twelve": 12,
+    "twelf": 12,
+    "thirteen": 13,
+    "fourteen": 14,
+    "fifteen": 15,
+    "sixteen": 16,
+    "seventeen": 17,
+    "eighteen": 18,
+    "nineteen": 19,
+    "twenty": 20,
+    "twentie": 20,
+    "thirty": 30,
+    "thirtie": 30,
+    "forty": 40,
+    "fortie": 40,
+    "fourty": 40,
+    "fourtie": 40,
+    "fifty": 50,
+    "fifdee": 50,
+    "fiddy": 50,
+    "fiftie": 50,
+    "sixty": 60,
+    "sixtie": 60,
+    "sixdee": 60,
+    "seventy": 70,
+    "seventie": 70,
+    "sevendie": 70,
```

Comment:
Have you seen the "die"/"dee" variants as ASR results, or are you aiming for exceptionally bad chat typing? In general, we've opted to include homonyms, but not typos.

Reply:
I just thought of those when I got to this part. I was trying to think of common variations, but I don't know if they are possible ASR results. My opinion is that it should be optional to use ASR, so we would want common ASR results and possible chat inputs.

Reply:
I agree, but spelling correction for chat is a much bigger project. You have to deal with character insertions/deletions/transpositions, etc. I don't think these are hurting anything, so they can stay, but in general we shouldn't spend too much effort on explicitly including typos in places like this unless they've been observed frequently, and including them gives us some performance benefit.

Reply:
That's a good point. I will remove what isn't included in the mobile libraries.